### PR TITLE
add geoinfo module and modify weather module

### PR DESCRIPTION
**Pull request by RMTT**
for convenient, the geoinfo can automatic get your wan ip and latitude,longitude, and then the weather module can automatic update, do not need to query your city id, but i also leave a  choice to use your own city id, just set the `auto_geoinfo = false`.
There are some other widgets can use those infomation, such as the dynamic wallpaper, the `geoinfo` module have already calculate the sunrise and sunset and solar noon time.

### CODE REVIEW

**Comment by eromatiya** [comment]
Nice! Thanks for this. Anyway, I can't review this right now. Been working to fix the lockscreen module atm.

File: config/awesome/surreal/configuration/secrets.lua [hunk 1]
```diff
@@ -39,7 +39,24 @@ return {
...
 		city_id       = '',
-		units         = 'metric'
+		units         = 'metric',
+		auto_geoinfo  = true
 
+	},
+
+	geoinfo = {
+		ipstack_key		= "",
+		proxy_for_check_ip = false,
```
Comment (eromatiya) [inline config/awesome/surreal/configuration/secrets.lua:49]: Finally, got a time to review this! Thank you for this as always! Anyway I have questions. 

Is the `ipstack_key` required or is it okay to just leave it empty? And please explain the other configurations like proxy_for_check_ip and etc. What do they do?

Also why are the items under the ***-- items which do not change*** in secrets.lua? Isn't the geoinfo module's the one that will set them?

**Review by eromatiya — COMMENTED** [review]
This is a nice addition but I think I'll just modify it a bit to improve the consistency of code across the setups.

**Comment by RMTT** [comment]
okay.
+ `ipstack_key` is required because i use it's api to get latitude and longitude
+ `proxy_for_check_ip` is true if you want to get ip via your proxy 
+ `proxy_for_ipstack` is true if you want request the ipstack api vis your proxy
+ `proxy_host` and `proxy_port` are used to set detailed proxy, just support http proxy, so don't need the `http://` prefix
+ `auto_geoinfo` used for weather widget, if it's true, you do not need fill the `city`


The items after `-- items which do not change` is used to save data, the `geoinfo` module will set thoes items after get and calculate those info from internet, and then other modules or widgets can use those info directly, maybe there is other better ways to do this, if you know something, can give me some tips : )

BTW, you just need fill  the `ipstack_key` in most envirionments

**Comment by RMTT** [comment]
The `system::network_connected` signal depends the network interface setting,if it's not corrrect, the signal will not be emitted, so i think we should split the interface setting to the configuration part, because it's a required setting.

**Comment by eromatiya** [comment]
Nice! Thanks for the detailed explanation! Uhm is `ipstack` the best service for this? I mean is there an alternative that is more simpler that will not require an account like `ifconfig.co` ? Edit so it turns out ifconfig.co doesn't return latitude and etc. haha. My bad

> So i think we should split the interface setting to the configuration part, because it's a required setting.

So you want to separate the interface setting/checking away from the network widget? Did I understand that correctly? The function that checks if the interface is correct/does exist is `check_network_mode()` in the network widget.

**Comment by RMTT** [comment]
> Nice! Thanks for the detailed explanation! Uhm is `ipstack` the best service for this? I mean is there an alternative that is more simpler that will not require an account like `ifconfig.co` ? Edit so it turns out ifconfig.co doesn't return latitude and etc. haha. My bad

Hmm, i have tried several apis, and the ipstack has best precision, and the free license is enough for personal use.


> So you want to separate the interface setting/checking away from the network widget? Did I understand that correctly? The function that checks if the interface is correct/does exist is `check_network_mode()` in the network widget.

Just place the interface setting to the configuration file,such as `secrets.lua`, i mean the setting of interface name

**Comment by eromatiya** [comment]
> Hmm, i have tried several apis, and the ipstack has best precision, and the free license is enough for personal use.

Yeah after researching ipstack is indeed the best for the job. 

> Just place the interface setting to the configuration file,such as secrets.lua, i mean the setting of interface name

I've been planning to create a configuration file. It's just I've been cleaning things up for now. Can this PR wait for now? I decided to merge this after implementing the configuration file for the widgets and modules. The problem is refactoring the code of the secrets, weather widget can cause conflicts here.

**Comment by RMTT** [comment]
okay, i think just need to change the `secrets.lua` to `config.lua` : )

**Comment by RMTT** [comment]
BTW, maybe you can move the config files to a external position, for example, place config files in `~/.the-golorious-dotfiles/`, and then we can update this project via aur directly, because the aur script will override the user's configuration folder now.

**Comment by eromatiya** [comment]
> okay, i think just need to change the secrets.lua to config.lua : )

Yep that's the plan! 

> BTW, maybe you can move the config files to a external position, for example, place config files in ~/.the-golorious-dotfiles/, and then we can update this project via aur directly, because the aur script will override the user's configuration folder now.

What if I create another repo for just the AwesomeWM setups? I mean this is my personal dotfiles and I think it's more cleaner if the setups have their own repo so the my personal design/philosophy and the community design/philosophy will not clash? And of course it will be easier to maintain the setups/wiki if they're standalone from this. What do you think?

**Comment by RMTT** [comment]
> I mean this is my personal dotfiles and I think it's more cleaner if the setups have their own repo so the my personal design/philosophy and the community design/philosophy will not clash?

Hmm, it's a great idea. You can just split the `config/awesome` to  a repo, i think it is enough.
The most import thing is split the configuration folder or files, it's convenient for users who want to update. For example, if the repo named `glorious-awesome`, you can create a folder in user's home foled named `.glorious-awesome`, and then put configuration files to this folder, and then in `rc.lua`, you can use `dofile` or `loadfile` to load those configuration files,and let them be a global vriable, so all modules and widgets can use those settings directly.

**Comment by eromatiya** [comment]
> Hmm, it's a great idea. You can just split the config/awesome to a repo,
i think it is enough.

Yep!

> For example, if the repo named glorious-awesome, you can create a folder
in user's home foled named .glorious-awesome, and then put configuration
files to this folder, and then in rc.lua, you can use dofile or loadfile to
load those configuration files,and let them be a global vriable, so all
modules and widgets can use those settings directly.

I think I'll just make it a global table. So you can just
`config.widget.email.key` for example. Where `config` is a global variable.

**Comment by RMTT** [comment]
> I think I'll just make it a global table. So you can just `config.widget.email.key` for example. Where `config` is a global variable.

hmm, i think use separated folder is good for users when update the project

**Comment by RMTT** [comment]
Hey, ang progress here?

**Comment by eromatiya** [comment]
The plan is to merge this after implementing the configuration file. But before adding the global confinguration, I'm cleaning up/fixing the widgets and inconsistencies in the setups.

**Comment by RMTT** [comment]
> The plan is to merge this after implementing the configuration file. But before adding the global confinguration, I'm cleaning up/fixing the widgets and inconsistencies in the setups.

hmm,i mean that split the awesome themes to a new repo : )

**Comment by eromatiya** [comment]
> hmm,i mean that split the awesome themes to a new repo : )

We need to finalize and finish everything here before migrating to a new repo. I think it's a cleaner way to migrate.

**Comment by RMTT** [comment]
okay : )

**Comment by eromatiya** [comment]
Sorry for the delay. Just been doing something lately. I think I'll just create a new repo for my dotfiles, then I'll rename this repo to something else like `awesome-glorious-configs` or something. This is better because it will preserved the contributions of the community.

**Comment by RMTT** [comment]
okay, i'm making [my own themes](https://github.com/RMTT/flyawesome), and found that split the configuration files to a stanalone file or folder seems difficult : (

**Comment by eromatiya** [comment]
Nice! Good luck!

**Comment by RMTT** [comment]
Closed this pull request, because i found a [better way](https://github.com/RMTT/flyawesome-python) to do this  : )

**Pull request closed without merging**